src/zarate.jl

```diff
@@ -147,6 +158,22 @@ function ordering_problem(layout::Zarate, graph, layer2nodes;
         end
     end
 
+    # force identical pairings in certain layers
+    for (layerA, layerB) in force_equal_layers  # ordering in layer1 and layer2 must be identical
+        nodesA = sort(layer2nodes[layerA]) # internal ordering in each layer matters
+        nodesB = sort(layer2nodes[layerB])
+        for (i1, (nA1, nB1)) in enumerate(zip(nodesA, nodesB))
+            for i2 in i1+1:length(nodesA)
+                nA2 = nodesA[i2]
+                nB2 = nodesB[i2]
+                variable_layerA = variable_by_name(m, "befores_$layerA[$nA1,$nA2]")
+                variable_layerB = variable_by_name(m, "befores_$layerB[$nB1,$nB2]")
```

Comment:
It would be nice not to do it using `variable_by_name`.
Can we change the earlier code to be a map that assembes a `Vector` of `befores` or something like that?

Reply:
Okay, after several subsequent fixes, I think the new approach works now.
It apparently introduced slight changes to the `sankey_3twos` reference image though.
I can update the reference images and verify correctness manually if you agree with the changes (but admittedly, it took more changes to avoid `variable_by_name` so I'd like to hear what you think).

Reply:
The reference issue seems to be related to the change in #32 to the `sankey_3twos` plot: on my side (and apparently on CI as well), I still get the original "unflipped" versions (and that's after I've rebased to main, i.e., with the HiGHS solver)